### PR TITLE
fix openblas thread version ivf build performance degradation

### PR DESCRIPTION
**Pull request by Presburger**
issue: #739

### CODE REVIEW

File: include/knowhere/comp/thread_pool.h [hunk 1]
```diff
@@ -239,10 +241,18 @@ class ThreadPool {
...
+            blas_thread_before = openblas_get_num_threads();
+            openblas_set_num_threads(1);
+#endif
+
             omp_set_num_threads(num_threads <= 0 ? omp_before : num_threads);
         }
         ~ScopedOmpSetter() {
             omp_set_num_threads(omp_before);
+#if defined(OPENBLAS_OS_LINUX)
+            omp_set_num_threads(blas_thread_before);
```
Comment (liliu-z) [inline include/knowhere/comp/thread_pool.h:254]: openblas_set_num_threads

**Review by liliu-z — CHANGES_REQUESTED** [review]

**Review by liliu-z — APPROVED** [review]
/lgtm

File: include/knowhere/comp/thread_pool.h [hunk 2]
```diff
@@ -239,10 +241,18 @@ class ThreadPool {
                 omp_before = build_pool_->size();
             }
 
+#if defined(OPENBLAS_OS_LINUX)
+            blas_thread_before = openblas_get_num_threads();
```
Comment (chasingegg) [inline include/knowhere/comp/thread_pool.h:245]: can we set the blas_thread_before same way as omp_before

**Pull request merged**